### PR TITLE
Feature ge medication basic and advanced search

### PR DESCRIPTION
**Pull request by denisfruza**
Basic search for medication - users can search medication by selecting medication name from the combo box. Result displays location of the medication and its quantity inside of that map object(room).
Advanced search for medication - users can click on selected map object and display its location on hospital map (jump to exact building and floor). 
For now, because medication doesn't have any connections with rooms in database, it is displayed ("stored") only in storage room and our product owner is okay with that.

### CODE REVIEW

**Comment by rammba** [comment]
[(0) PR Feature ge medication basic and advanced search](https://trello.com/c/LB8B5eQn/109-0-pr-feature-ge-medication-basic-and-advanced-search)

File: WPFHospitalEditor/SearchResultDialog.xaml.cs [hunk 1]
```diff
@@ -338,6 +351,20 @@ private string[] EquipmentToContentRows()
             }
             return contentRows;
         }
+        private string[] MedicationToContentRows()
+        {
+            string[] contentRows = new string[HospitalMap.medicationSearchResult.Count()];
+            for (int i = 0; i < HospitalMap.medicationSearchResult.Count(); i++)
+            {
+                MapObject mo = mapObjectController.findMapObjectById(STORAGEROOM_ID);
```
Comment (BoKna031) [inline WPFHospitalEditor/SearchResultDialog.xaml.cs:359]: This should be out of loop because it is constant value for each iteration.

**Review by BoKna031 — COMMENTED** [review]
Methods setWindowForMediaction(forEquipment) and methods Equipment(Medication)ToContentRows  are almost same. You could refactor that methods to be one. There would be probably be one if statement to define which logic you should use.

File: WPFHospitalEditor/SearchResultDialog.xaml.cs (hunk 1 above)
Comment (denisfruza) [reply WPFHospitalEditor/SearchResultDialog.xaml.cs:359]: Storage room finding is now out of the loop.

**Comment by denisfruza** [comment]
Refactored methods that were almost the same- setWindowForMediaction(forEquipment) and  Equipment(Medication)ToContentRows

File: WPFHospitalEditor/SearchResultDialog.xaml.cs [hunk 2]
```diff
@@ -300,12 +300,14 @@ private void SetContentRowsAndColumnsNumber()
...
-                        setWindowForEquipment();
+                        setWindowForEquipmentAndMedication();
+                        contentRows = EquipmentToContentRows();
                     }
                     break;
                 case SearchType.MedicationSearch:
                     {
-                        setWindowForMedication();
+                        setWindowForEquipmentAndMedication();
+                        contentRows = MedicationToContentRows();
```
Comment (BoKna031) [inline WPFHospitalEditor/SearchResultDialog.xaml.cs]: I think that is better to use if else statements. Also with that you could reduce number of lines in code (you could write setWindowForEquipmentAndMedication() just one time )

**Review by BoKna031 — APPROVED** [review]

**Review by rammba — APPROVED** [review]

**Pull request merged**